cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -102,9 +130,18 @@ type nodeTemplate struct {
 	Taints       []apiv1.Taint
 }
 
+func init() {
+	controlBurst = flag.Int("control-apiserver-burst", rest.DefaultBurst, "Throttling burst configuration for the client to control cluster's apiserver.")
+	controlQPS = flag.Float64("control-apiserver-qps", float64(rest.DefaultQPS), "Throttling QPS configuration for the client to control cluster's apiserver.")
+	targetBurst = flag.Int("target-apiserver-burst", rest.DefaultBurst, "Throttling burst configuration for the client to target cluster's apiserver.")
+	targetQPS = flag.Float64("target-apiserver-qps", float64(rest.DefaultQPS), "Throttling QPS configuration for the client to target cluster's apiserver.")
+}
+
 func createMCMManagerInternal(discoveryOpts cloudprovider.NodeGroupDiscoveryOptions) (*McmManager, error) {
+	const resyncPeriod = 12 * time.Hour
```

Comment:
I suspect this is too large. kubelet for example [uses a 15 minute resync period](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubeletconfig/watch.go#L42), same with kube-proxy. It is configurable in kube-proxy as `config-sync-period`, so maybe we want the same here. I think the issue is that watches can silently drop in some configurations (for example, if you're routing apiserver traffic via a proxy like an AWS ELB which happen to restart/die). The resync is necessary to detect these stale watches.

Reply:
The concern you bring to the table is a valid one. Thanks for this. I looked through several controllers including the [kube-controller-manager](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/), Kubelet, kube-proxy and other controllers. The general trend I see is that the cache resync period is configured to controllers to 12hours including the KCM in the link above. However, like you have mentioned for kubelet and kube-proxy the values are 15mins. 

I agree with your concern about the stale cache however, resync of cache every 15mins seems CPU intensive to me. I even checked other Gardener controllers including the MCM that seems to be using 12 hours and working without problems until now. 

Finally, I looked at the CA inbuilt caches, and looks like they are using a [static 1hour](https://github.com/gardener/autoscaler/blob/machine-controller-manager-provider/cluster-autoscaler/utils/kubernetes/listers.go#L246). So keeping it aligned with the other parts of CA how about we align it to default to 1hour for now. And as always this flag is configurable and can be tweaked if we find it suboptimal in the future?

Reply:
Ah you're right. I wonder why it's 12h in controller manager. Seems like we looked at opposite things :). Maybe it's not an issue in that case.

Reply:
It's probably because controller manager has many more watches potentially than kubelet or kube-proxy. That's my guess at least.

Reply:
Indeed :) I have taken a middle ground of `1hour` for now. Also considering the cluster autoscaler configs above. I hope it seems reasonable for now.